connectivity/check/metricssource.go

```diff
@@ -0,0 +1,56 @@
+package check
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/components"
+)
+
+const prometheusContainerPortName = "prometheus"
+
+// MetricsSource defines the info for a source to be used in metrics collection.
+type MetricsSource struct {
+	Name string // the name of the source, e.g.: cilium-agent
+	Pods []Pod  // the list of pods for the given source
+	Port string // the container port value for prometheus
+}
+
+// CiliumAgentMetrics returns the MetricsSource for the cilium-agent component.
+func (ct *ConnectivityTest) CiliumAgentMetrics() MetricsSource {
+	ciliumPods := ct.CiliumPods()
+	if len(ciliumPods) == 0 {
+		return MetricsSource{}
+	}
+
+	var pods []Pod
+	for _, p := range ciliumPods {
+		pods = append(pods, p)
+	}
+
+	source := MetricsSource{
+		Name: components.CiliumAgentName,
+		Pods: pods,
+	}
+
+	// Retrieve the container port value for Prometheus.
+	for _, p := range ciliumPods {
+		// parse all the containers
+		for _, c := range p.Pod.Spec.Containers {
+			if c.Name == components.CiliumAgentName {
+				// parse all the container ports
+				for _, port := range c.Ports {
+					if port.Name == prometheusContainerPortName {
+						source.Port = fmt.Sprintf("%d", port.ContainerPort)
```

Comment:
consider exiting the `range ciliumPods` loop early here once we found a port - alternatively, it might be cleaner to actually `(pod, port)` pairs in the `MetricsSouce`.

Reply:
I added the early exit, but I will be happy to discuss the interest of pairing pod, port :)

Reply:
For the record, we decided with @gandro to not pair a pod to a port for the moment but we could have a usecase in the future (e.g. cluster mesh use-case: different configuration of ports for a component through different clusters)